grails-app/services/org/pih/warehouse/core/OrganizationIdentifierService.groovy

```diff
@@ -0,0 +1,73 @@
+package org.pih.warehouse.core
+
+import grails.core.GrailsApplication
+import org.apache.commons.lang.WordUtils
+
+// TODO: Try to merge this logic into IdentifierService. We'd need to support a new ${abbreviation} keyword in the
+//       format that would get injected to based on the same logic that we have here. Then we'd add a new
+//       identifier.x.abbreviation.field = "path.to.field" config that we can use to extract the field from the entity.
+//       Or if that's too challenging, we can simply return the string to use via abstract method in IdentifierService.
+class OrganizationIdentifierService {
```

Comment:
A majority of this code is just copied out from IdentifierService and some other places. I couldn't get around to refactoring this as a part of this change because it has such a complex configuration.

All I really did here was add comments to clarify the code.

See the TODO here for how we could refactor in the future.

Reply:
The TODO describes my goal here. I want to have a single generate() method that is driven entirely by properties. I'm imagining this being configured like:
```
openboxes.identifier.organization.format = "\${abbreviation}"
openboxes.identifier.organization.abbreviation.field = "name"
openboxes.identifier.organization.minSize = 2
openboxes.identifier.organization.maxSize = 3
```
then run with:
```
Organization org = new Organization(name: "A Good Name")
String id = organizationIdentifierService.generate(org)
id == "AGN"
```